### PR TITLE
docs: Added artifact.md to docs

### PR DESCRIPTION
**Pull request by sajayantony**
Added artifact.md and updated the media-types diagram dot and png.

Signed-off-by: Sajay Antony <sajaya@microsoft.com>

### CODE REVIEW

File: img/media-types.dot [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
+    }
+    config [shape=note, label="Image config JSON\napplication/vnd.oci.image.config.v1+json"]
     layer [shape=note, label="Layer tar archive\napplication/vnd.oci.image.layer.v1.tar\napplication/vnd.oci.image.layer.v1.tar+gzip\napplication/vnd.oci.image.layer.nondistributable.v1.tar\napplication/vnd.oci.image.layer.nondistributable.v1.tar+gzip"]
   }
 
   imageIndex -> imageIndex [label="1..*"]
   imageIndex -> manifest [label="1..*"]
   manifest -> config [label="1..1"]
   manifest -> layer [label="1..*"]
+  artifact -> manifest [label="1..1"] [constraint = false];
```
Comment (sudo-bmitch) [inline img/media-types.dot:17]: If this is capturing the "subject" field, then we also need two more self references like there is for `imageIndex`, and the reference is also optional, so `0..1`:

```suggestion
  artifact -> manifest [label="0..1"] [constraint = false];
  artifact -> artifact [label="0..1"];
  manifest -> manifest [label="0..1"];
```

I'm not sure how to capture that it's `0..1` for multiple targets and you can't have a manifest with both a subject of another manifest and an artifact, but that may be more detail than needed.

File: img/media-types.dot [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
+  artifact -> imageIndex [label="1..1"]
+  artifact -> layer[label="1..*"]
```
Comment (sudo-bmitch) [inline img/media-types.dot:19]: Since blobs are optional for artifacts, I think we want `0..*` here.

```suggestion
  artifact -> layer[label="0..*"]
```

**Review by sudo-bmitch — CHANGES_REQUESTED** [review]
Thanks for running with this one. We also need to add a reference to `artifact.md` in the "Table of Contents" at the top of `spec.md`.

**Comment by sudo-bmitch** [comment]
The more I think about the subject field in the graphic, and anything with a descriptor to a manifest, the more I want a general manifest and blob in the graphic, and have pointers go to those general types. I'll do some digging into graphviz and see if there's a better way to draw it.

In the mean time, if another wants to approve this, I'll update my review and follow up with a separate PR.

**Review by sudo-bmitch — APPROVED** [review]
LGTM. There's still some work to do on the graphic, but this gets us in a better place.

**Review by jonjohnsonjr — APPROVED** [review]

**Pull request merged**